woudc_data_registry/epicentre/instrument.py

```diff
@@ -67,18 +69,25 @@ def build_instrument(ecsv):
     serial = str(ecsv.extcsv['INSTRUMENT']['Number'])
     station = str(ecsv.extcsv['PLATFORM']['ID'])
     dataset = ecsv.extcsv['CONTENT']['Category']
+    contributor = str(ecsv.extcsv['DATA_GENERATION']['Agency'])
+    project = str(ecsv.extcsv['CONTENT']['Class'])
     location = [ecsv.extcsv['LOCATION'].get(f, None)
                 for f in ['Longitude', 'Latitude', 'Height']]
     timestamp_date = ecsv.extcsv['TIMESTAMP']['Date']
 
-    instrument_id = ':'.join([name, model, serial, station, dataset])
+    contributor_id = ':'.join([contributor, project])
+
+    instrument_id = ':'.join([name, model, serial,
+                              station, dataset, contributor])
+
     model = {
         'identifier': instrument_id,
         'name': name,
         'model': model,
         'serial': serial,
         'station_id': station,
         'dataset_id': dataset,
+        'contributor_id': contributor_id,
```

Comment:
This "model" dict needs to match up with the keys that `Instrument.__init__` expects. That means `contributor_id` should be passed as `contributor` and `project`, as specified there.